docs/architecture/blazor-for-web-forms-developers/state-management.md

```diff
@@ -1,23 +1,89 @@
 
-<!--
-- View state
-- Session state
-- Local storage
-- App state
--->
+## Request state management with ViewState
+
+When discussing state management regarding a Web Forms application, many developers will immediately think of ViewState. In Web Forms, ViewState managed the state of the content between HTTP requests by sending a large encoded block of text back and forth to the browser. The ViewState field could be overwhelmed with content from a page containing many elements, expanding as large as several megabytes in size.
+
+With Blazor Server, each requested component's state is persisted in memory on the server between requests. State will only be disposed when the user navigates away from the component. All in-scope field values of the component are available between interactions with the server.
+
+There are several advantages of this feature:
+
+- Component's state is always known and not rebuilt between requests.
+- State isn't transmitted to the browser.
+
+However, there are some disadvantages to in-memory component state persistence to be aware of:
+
+- If the server restarts between request, state is lost.
+- Your application web server load-balancing solution must include sticky sessions to ensure that all requests from the same browser return to the same server. If a request goes to a different server, state will be lost.
+- Persistence of component state on the server can lead to memory pressure on the web server.
+
+For the preceding reasons, don't rely on just the state of the component to reside in-memory on the server. Your application should also include some backing data store for data between requests. Some simple examples of this strategy:
+
+- In a shopping cart application, persist the content of new items added to the cart in a database record. If the state on the server is lost, you can reconstitute it from the database records.
+- In a multi-part web form, your users will expect your application to remember values between each request. Write the data between each of your user's posts to a data store so that they can be fetched and assembled into the final form response structure when the multi-part form is completed.
+
+The signature of the .NET Framework `Session` object isn't the same as the ASP.NET Core `Session` object. Consider [the documentation for the new ASP.NET Core Session](/dotnet/api/microsoft.aspnetcore.http.isession) before deciding to migrate and use the new session state feature.
```

Comment:
This looks like it would fit better in the session state section below.

Reply:
Relocated the sentence to the Session State section